### PR TITLE
Dynamic cmap

### PR DESCRIPTION
**Pull request by CJ-Wright**
I have started doing some plumbing to allow for the default values of the stack explorer to be set during its initial call.

### CODE REVIEW

File: xray_vision/messenger/mpl/cross_section_2d.py [hunk 1]
```diff
@@ -60,7 +60,8 @@ class CrossSection2DMessenger(AbstractMessenger2D, AbstractMPLMessenger):
     to pass commands down to the gui-independent layer
     """
 
-    def __init__(self, data_list, key_list, parent=None, *args, **kwargs):
+    def __init__(self, data_list, key_list, parent=None, start_cmap=None,
```
Comment (tacaswell) [inline xray_vision/messenger/mpl/cross_section_2d.py]: just call it `cmap`

**Comment by CJ-Wright** [comment]
How are the defaults set for the normalization and range?

**Comment by ericdill** [comment]
That is a good question... Let me re-learn this code and answer that for you

**Comment by ericdill** [comment]
`range` is the intensity range of the image?

**Comment by CJ-Wright** [comment]
Yes I think so

**Comment by ericdill** [comment]
Ok, see cj-wright/xray-vision#1 for a minor tweak to set the range.  You can add something like this:
```python
        self._ctrl_widget.set_image_intensity_behavior('percentile')
        self._ctrl_widget.set_limits(25, 75)
```

to the `__init__` of `CrossSectionMainWindow` and then pass in those values as kwargs or something

**Comment by ericdill** [comment]
looking at the log norm now

**Comment by CJ-Wright** [comment]
It seems that even though the cmap box says the correct map, that map is not implemented. Do you know what could be happening? @ericdill

**Comment by ericdill** [comment]
Let me investigate

**Comment by ericdill** [comment]
@CJ-Wright check out cj-wright/xray-vision#2... That should fix these problems.

**Comment by CJ-Wright** [comment]
Awesome, it works and also fixed the bug which caused it to not load data at the start! Thank  you very much @ericdill

**Comment by CJ-Wright** [comment]
I think this might be ready for review/safe to merge.
@tacaswell @ericdill

**Comment by tacaswell** [comment]
:+1:   I only gave it a cursory read, but nothing jumped out at me.  If it works, it is probably good to merge.

**Comment by ericdill** [comment]
Awesome, thanks @tacaswell.  Thanks for the effort on this @CJ-Wright; your code + desire to see this functionality makes the cross section widget much more useful.

**Pull request merged**